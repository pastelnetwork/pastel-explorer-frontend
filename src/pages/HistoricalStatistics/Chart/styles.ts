import { makeStyles } from '@material-ui/core/styles';
import { TAppTheme } from '@theme/index';

export const eChartLineStyles = makeStyles((theme: TAppTheme) => ({
  container: {
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    height: '100%',
    width: '100%',
    padding: '0 20px',
    textAlign: 'center',
    boxSizing: 'border-box',
    position: 'relative',
    color: 'rgb(171, 170, 193)',
    [theme.breakpoints.down('sm')]: {
      padding: 0,
    },
  },
  lineChartWrap: {
    width: '100%',
    height: 'auto',
    marginBottom: 20,
  },
  lineChartContainer: {
    display: 'block',
    width: '100%',
    height: '100%',
    backgroundColor: '#212121',
  },
  lineChartTitle: {
    width: '100%',
    textAlign: 'left',
    fontSize: '30px',
    [theme.breakpoints.down('sm')]: {
      fontSize: 15,
    },
  },
  lineChartHeader: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-between',
    paddingTop: '30px',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      paddingTop: 10,
      paddingLeft: 10,
      justifyContent: 'flex-start',
      flexDirection: 'column',
    },
  },
  uploadButton: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: '#6151d0',
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '14px',
    padding: '8px 16px',
    borderRadius: '4px',
    marginLeft: '8px',
    marginRight: '8px',
    '&:hover': {
      cursor: 'pointer',
      opacity: 0.8,
    },
  },
  reactECharts: {
    width: '100%',
    height: 'calc(100vh - 450px) !important',
  },
  granularitySelect: {
    display: 'flex',
    width: '100%',
    maxWidth: 200,
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15,
    marginRight: 50,
  },
  periodSelect: {
    display: 'flex',
    minWidth: '200px',
    // justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: '15px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 5,
    },
  },
  lineChartDownloadButtonBar: {
    display: 'flex',
    width: '100%',
    justifyContent: 'flex-end',
    height: '30px',
    [theme.breakpoints.down('sm')]: {
      marginTop: 10,
      justifyContent: 'flex-start',
    },
  },
  lineChartFooter: {
    display: 'flex',
    width: '100%',
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingBottom: 30,
    [theme.breakpoints.down('sm')]: {
      flexWrap: 'wrap',
      justifyContent: 'flex-start',
      paddingBottom: 10,
    },
  },
  lineChartThemeSelect: {
    display: 'flex',
    minWidth: '150px',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  themeSelectButton: {
    width: '30px',
    height: '30px',
    borderRadius: '5px',
  },
  filterButton: {
    backgroundColor: '#eee',
    color: '#212124',
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: '12px',
    padding: '4px 8px',
    borderRadius: '3px',
    borderWidth: '1px',
    borderColor: '#212124',
    marginLeft: '4px',
    marginRight: '4px',
    '&:hover': {
      cursor: 'pointer',
      opacity: '0.8',
    },
  },
  activeButton: {
    backgroundColor: '#c3921f',
    borderColor: '#c3921f',
    color: '#fff',
  },
  activeThemeButton: {
    borderColor: '#008000',
    borderWidth: '5px',
  },
}));
