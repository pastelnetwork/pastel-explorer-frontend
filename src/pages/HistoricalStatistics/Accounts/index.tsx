import { useEffect, useState } from 'react';

import { PeriodTypes, transformAccountDataChart } from '@utils/helpers/statisticsLib';
import { periods, info, cacheList } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { readCacheValue, setCacheValue } from '@utils/helpers/localStorage';
import { TLineChartData } from '@utils/types/IStatistics';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout/HistoricalStatisticsLayout';
import useAccounts from '@hooks/useAccounts';
import { translate } from '@utils/helpers/i18n';

import { EChartsLineChart } from '../Chart/EChartsLineChart';

function Accounts() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const [isLoading, setLoading] = useState(false);
  const swrData = useAccounts(period);

  useEffect(() => {
    let currentCache = readCacheValue(cacheList.accounts) || {};
    if (currentCache[period]) {
      setChartData(currentCache[period].parseData as TLineChartData);
      setLoading(false);
    } else {
      setLoading(true);
    }
    if (!swrData.isLoading && swrData.data) {
      const parseData = transformAccountDataChart(swrData.data, period, '');
      setChartData(parseData);
      currentCache = {
        ...currentCache,
        [period]: {
          parseData,
        },
      };
      setCacheValue(
        cacheList.accounts,
        JSON.stringify({
          currentCache,
          lastDate: Date.now(),
        }),
      );
      setLoading(false);
    }
  }, [period, swrData.isLoading, swrData.data]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout
      currentBgColor={currentBgColor}
      title={translate('pages.historicalStatistics.accounts')}
    >
      <EChartsLineChart
        chartName="accounts"
        dataX={chartData?.dataX}
        dataY={chartData?.dataY}
        title={translate('pages.historicalStatistics.accounts')}
        info={info}
        period={period}
        offset={10}
        periods={periods[6]}
        handleBgColorChange={handleBgColorChange}
        handlePeriodFilterChange={handlePeriodFilterChange}
        setHeaderBackground
        isLoading={isLoading}
      />
    </HistoricalStatisticsLayout>
  );
}

export default Accounts;
