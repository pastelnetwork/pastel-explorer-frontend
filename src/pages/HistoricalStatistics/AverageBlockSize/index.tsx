// react
import { useEffect, useState } from 'react';
// third party
import { Skeleton } from '@material-ui/lab';
// application
import { TLineChartData, TAverageBlockSize } from '@utils/types/IStatistics';
import { TGranularity, PeriodTypes, transformAverageBlockSize } from '@utils/helpers/statisticsLib';
import * as URLS from '@utils/constants/urls';
import { useFetch } from '@utils/helpers/useFetch/useFetch';
import {
  BLOCK_CHART_DEFAULT_GRANULARITY,
  granularities,
  periods,
  info,
} from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';
import { EChartsLineChart } from '../Chart/EChartsLineChart';

const AverageBlockSize = (): JSX.Element => {
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const [granularity, setGranularity] = useState<TGranularity>(BLOCK_CHART_DEFAULT_GRANULARITY);
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const fetchStats = useFetch<{ data: Array<TAverageBlockSize> }>({
    method: 'get',
    url: URLS.GET_STATISTICS_AVERAGE_BLOCK_SIZE,
  });
  useEffect(() => {
    const loadLineChartData = async () => {
      const data = await fetchStats.fetchData({
        params: { sortDirection: 'DESC', period, granularity },
      });
      if (data) {
        const parseData = transformAverageBlockSize(data.data);
        setChartData(parseData);
      }
    };
    loadLineChartData();
  }, [granularity, period]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  const handleGranularityFilterChange = (value: TGranularity) => {
    setGranularity(value);
  };

  return (
    <HistoricalStatisticsLayout currentBgColor={currentBgColor}>
      {chartData ? (
        <EChartsLineChart
          chartName="averageblocksize"
          dataX={chartData?.dataX}
          dataY={chartData?.dataY}
          title="Average Block Size (kB)"
          info={info}
          offset={1}
          period={period}
          granularity={granularity}
          granularities={granularities[0]}
          periods={periods[1]}
          handleBgColorChange={handleBgColorChange}
          handlePeriodFilterChange={handlePeriodFilterChange}
          handleGranularityFilterChange={handleGranularityFilterChange}
        />
      ) : (
        <Skeleton animation="wave" variant="rect" height={386} />
      )}
    </HistoricalStatisticsLayout>
  );
};

export default AverageBlockSize;
