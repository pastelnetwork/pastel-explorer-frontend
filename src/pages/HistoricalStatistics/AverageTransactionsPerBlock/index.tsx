// react
import { useEffect, useState } from 'react';
// third party
import { Skeleton } from '@material-ui/lab';
// application
import * as URLS from '@utils/constants/urls';
import { useFetch } from '@utils/helpers/useFetch/useFetch';
import { PeriodTypes, transformCharts } from '@utils/helpers/statisticsLib';
import { periods, info } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { TLineChartData, TTransactionsChart } from '@utils/types/IStatistics';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';
import { EChartsLineChart } from '../Chart/EChartsLineChart';

function AverageTransactionsPerBlock() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const fetchStats = useFetch<{ data: Array<TTransactionsChart> }>({
    method: 'get',
    url: URLS.GET_BLOCKS_CHARTS,
  });
  useEffect(() => {
    const loadLineChartData = async () => {
      const data = await fetchStats.fetchData({
        params: { period, sortDirection: 'DESC', func: 'AVG', col: 'transactionCount' },
      });
      if (data) {
        const parseData = transformCharts(data.data);
        setChartData(parseData);
      }
    };
    loadLineChartData();
  }, [period]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout currentBgColor={currentBgColor}>
      {chartData ? (
        <EChartsLineChart
          chartName="transactionfee"
          dataX={chartData?.dataX}
          dataY={chartData?.dataY}
          title="Average Transaction Per Block"
          info={info}
          offset={1}
          period={period}
          periods={periods[1]}
          handleBgColorChange={handleBgColorChange}
          handlePeriodFilterChange={handlePeriodFilterChange}
        />
      ) : (
        <Skeleton animation="wave" variant="rect" height={386} />
      )}
    </HistoricalStatisticsLayout>
  );
}

export default AverageTransactionsPerBlock;
