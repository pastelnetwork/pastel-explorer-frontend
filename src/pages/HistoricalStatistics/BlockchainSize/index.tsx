// react
import { useEffect, useState } from 'react';
// third party
import { Skeleton } from '@material-ui/lab';
// application
import * as URLS from '@utils/constants/urls';
import { useFetch } from '@utils/helpers/useFetch/useFetch';
import { PeriodTypes, transformBlockchainSize } from '@utils/helpers/statisticsLib';
import { periods, info } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { TLineChartData, TTransactionsChart } from '@utils/types/IStatistics';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';

import { EChartsLineChart } from '../Chart/EChartsLineChart';

function BlockchainSize() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const fetchStats = useFetch<{ data: Array<TTransactionsChart> }>({
    method: 'get',
    url: URLS.GET_BLOCKS_CHARTS,
  });
  useEffect(() => {
    const loadLineChartData = async () => {
      const data = await fetchStats.fetchData({
        params: { period, sortDirection: 'DESC', func: 'SUM', col: 'size' },
      });
      if (data) {
        const parseData = transformBlockchainSize(data.data);
        setChartData(parseData);
      }
    };
    loadLineChartData();
  }, [period]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout currentBgColor={currentBgColor}>
      {chartData ? (
        <EChartsLineChart
          chartName="transactionfee"
          dataX={chartData?.dataX}
          dataY={chartData?.dataY}
          title="Blockchain Size (Mb)"
          info={info}
          period={period}
          offset={1}
          periods={periods[1]}
          handleBgColorChange={handleBgColorChange}
          handlePeriodFilterChange={handlePeriodFilterChange}
        />
      ) : (
        <Skeleton animation="wave" variant="rect" height={386} />
      )}
    </HistoricalStatisticsLayout>
  );
}

export default BlockchainSize;
