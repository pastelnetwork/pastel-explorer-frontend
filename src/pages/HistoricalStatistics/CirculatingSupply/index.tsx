import { useEffect, useState } from 'react';

import { PeriodTypes, transformStatisticsChart } from '@utils/helpers/statisticsLib';
import { periods, info, cacheList } from '@utils/constants/statistics';
import { useBackgroundChart } from '@utils/hooks';
import { readCacheValue, setCacheValue } from '@utils/helpers/localStorage';
import { TLineChartData } from '@utils/types/IStatistics';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';
import { getCurrencyName } from '@utils/appInfo';
import useCirculatingSupply from '@hooks/useCirculatingSupply';

import { EChartsLineChart } from '../Chart/EChartsLineChart';

function CirculatingSupply() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const [isLoading, setLoading] = useState(false);
  const swrData = useCirculatingSupply(period);

  useEffect(() => {
    let currentCache = readCacheValue(cacheList.circulatingSupply) || {};
    if (currentCache[period]) {
      setChartData(currentCache[period].parseData as TLineChartData);
      setLoading(false);
    } else {
      setLoading(true);
    }
    if (!swrData.isLoading && swrData.data) {
      const parseData = transformStatisticsChart(swrData.data, period, '');
      setChartData(parseData);
      currentCache = {
        ...currentCache,
        [period]: {
          parseData,
        },
      };
      setCacheValue(
        cacheList.circulatingSupply,
        JSON.stringify({
          currentCache,
          lastDate: Date.now(),
        }),
      );
      setLoading(false);
    }
  }, [period, swrData.isLoading, swrData.data]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout
      currentBgColor={currentBgColor}
      title={`Circulating Supply (${getCurrencyName()})`}
    >
      <EChartsLineChart
        chartName="circulatingSupply"
        dataX={chartData?.dataX}
        dataY={chartData?.dataY}
        title={`Circulating Supply (${getCurrencyName()})`}
        info={info}
        period={period}
        offset={0}
        periods={periods[6]}
        handleBgColorChange={handleBgColorChange}
        handlePeriodFilterChange={handlePeriodFilterChange}
        setHeaderBackground
        isLoading={isLoading}
      />
    </HistoricalStatisticsLayout>
  );
}

export default CirculatingSupply;
