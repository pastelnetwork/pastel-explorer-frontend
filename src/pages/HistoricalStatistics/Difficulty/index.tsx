// react
import { useEffect, useState } from 'react';
// third party
import { Skeleton } from '@material-ui/lab';
// application
import * as URLS from '@utils/constants/urls';
import { useFetch } from '@utils/helpers/useFetch/useFetch';
import { PeriodTypes, transformDifficultyInfo } from '@utils/helpers/statisticsLib';
import { periods, info } from '@utils/constants/statistics';
import { IStatistic, TLineChartData } from '@utils/types/IStatistics';
import { useBackgroundChart } from '@utils/hooks';
import HistoricalStatisticsLayout from '@components/HistoricalStatisticsLayout';
import { EChartsLineChart } from '../Chart/EChartsLineChart';

function Difficulty() {
  const [chartData, setChartData] = useState<TLineChartData | null>(null);
  const [currentBgColor, handleBgColorChange] = useBackgroundChart();
  const [period, setPeriod] = useState<PeriodTypes>(periods[1][0]);
  const fetchStats = useFetch<{ data: Array<IStatistic> }>({
    method: 'get',
    url: URLS.GET_STATISTICS,
  });
  useEffect(() => {
    const loadLineChartData = async () => {
      const data = await fetchStats.fetchData({
        params: { sortDirection: 'DESC', period },
      });
      if (data) {
        const parseData = transformDifficultyInfo(data.data);
        setChartData(parseData);
      }
    };
    loadLineChartData();
  }, [period]);

  const handlePeriodFilterChange = (value: PeriodTypes) => {
    setPeriod(value);
  };

  return (
    <HistoricalStatisticsLayout currentBgColor={currentBgColor}>
      {chartData ? (
        <EChartsLineChart
          chartName="difficulty"
          dataX={chartData?.dataX}
          dataY={chartData?.dataY}
          title="Network Difficulty"
          period={period}
          info={info}
          offset={10000}
          periods={periods[1]}
          handleBgColorChange={handleBgColorChange}
          handlePeriodFilterChange={handlePeriodFilterChange}
        />
      ) : (
        <Skeleton animation="wave" variant="rect" height={386} />
      )}
    </HistoricalStatisticsLayout>
  );
}

export default Difficulty;
