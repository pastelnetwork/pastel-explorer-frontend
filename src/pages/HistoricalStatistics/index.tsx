import Grid from '@material-ui/core/Grid';
import { Link } from 'react-router-dom';
import { statistics } from '@utils/constants/statistics';
import { Title, Wrapper } from './StatisticsOvertime.styles';

const Statistics = () => {
  return (
    <Wrapper>
      <Title>Pastel Statistics</Title>
      <svg
        width="19"
        height="17"
        viewBox="0 0 19 17"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.0938 7.71894H2.5082L8.47617 2.03457C8.78867 1.73691 8.80078 1.24238 8.50313 0.929878C8.20586 0.617769 7.71133 0.605269 7.39844 0.902925L0.582812 7.39511C0.287891 7.69042 0.125 8.08261 0.125 8.50019C0.125 8.91738 0.287891 9.30996 0.596484 9.61816L7.39883 16.0971C7.55 16.2412 7.74375 16.3127 7.9375 16.3127C8.14375 16.3127 8.35 16.2314 8.50352 16.0701C8.80117 15.7576 8.78906 15.2635 8.47656 14.9658L2.48359 9.28144H18.0938C18.525 9.28144 18.875 8.93144 18.875 8.50019C18.875 8.06894 18.525 7.71894 18.0938 7.71894Z"
          fill="white"
          stroke="white"
          strokeWidth="0.2"
        />
      </svg>
      <Grid container spacing={6} alignItems="center">
        {statistics.map(({ id, image, title, url }) => (
          <Grid item key={id} className="card-item" xs={12} sm={6} lg={4} xl={3}>
            <Link to={url}>
              <img alt="" width="100%" height="100%" style={{ objectFit: 'contain' }} src={image} />
              <h2>{title}</h2>
            </Link>
          </Grid>
        ))}
      </Grid>
    </Wrapper>
  );
};

export default Statistics;
