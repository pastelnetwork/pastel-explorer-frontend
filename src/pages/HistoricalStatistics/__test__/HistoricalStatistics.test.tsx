import { shallow } from 'enzyme';
import Grid from '@material-ui/core/Grid';
import { Link } from 'react-router-dom';
import 'jest-styled-components';

import i18next from '../../../utils/helpers/i18n';
import HistoricalStatistics from '../index';
import * as Styles from '../StatisticsOvertime.styles';

jest.mock('i18next-http-backend');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () =>
          new Promise(() => {
            // noop
          }),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {
      // noop
    },
  },
}));
i18next.t = jest.fn().mockImplementation((...arg) => {
  return arg[0];
});

describe('pages/HistoricalStatistics', () => {
  const wrapper = shallow(<HistoricalStatistics />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <Grid>', () => {
    expect(wrapper.find(Grid).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Link>', () => {
    expect(wrapper.find(Link).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.Wrapper>', () => {
    expect(wrapper.find(Styles.Wrapper).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.ContentWrapper>', () => {
    expect(wrapper.find(Styles.ContentWrapper).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.CardItem>', () => {
    expect(wrapper.find(Styles.CardItem).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.BlockTitle>', () => {
    expect(wrapper.find(Styles.BlockTitle).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <Styles.ChartImage>', () => {
    expect(wrapper.find(Styles.ChartImage).length).toBeGreaterThanOrEqual(1);
  });

  test('should render <img>', () => {
    expect(wrapper.find('img').length).toBeGreaterThanOrEqual(1);
  });
});
