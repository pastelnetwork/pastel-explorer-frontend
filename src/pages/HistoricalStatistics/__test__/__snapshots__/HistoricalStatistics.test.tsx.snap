// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/HistoricalStatistics renders correctly 1`] = `ShallowWrapper {}`;
